src/vs/platform/windows/electron-main/windowsMainService.ts

```diff
@@ -1498,27 +1501,31 @@ export class WindowsMainService extends Disposable implements IWindowsMainServic
 			}
 		}
 
+		if (this.userDataProfilesMainService.isEnabled()) {
+			const workspace = configuration.workspace ?? toWorkspaceIdentifier(configuration.backupPath, false);
+			const profile = await this.resolveProfileForBrowserWindow(options, workspace);
+			configuration.profiles.profile = profile;
+
+			if (!configuration.extensionDevelopmentPath) {
+				// Associate the configured profile to the workspace
+				// unless the window is for extension development,
+				// where we do not persist the associations
+				await this.userDataProfilesMainService.setProfileForWorkspace(workspace, profile);
+			}
+		}
+
 		// Load it
 		window.load(configuration);
 	}
 
-	private async resolveProfileForBrowserWindow(options: IOpenBrowserWindowOptions): Promise<IUserDataProfile> {
-		let profile: IUserDataProfile | undefined;
-		if (this.userDataProfilesMainService.isEnabled()) {
-			if (options.forceProfile) {
-				profile = this.userDataProfilesMainService.profiles.find(p => p.name === options.forceProfile) ?? await this.userDataProfilesMainService.createNamedProfile(options.forceProfile);
-			} else if (options.forceTempProfile) {
-				profile = await this.userDataProfilesMainService.createTransientProfile();
-			}
+	private async resolveProfileForBrowserWindow(options: IOpenBrowserWindowOptions, workspace: IAnyWorkspaceIdentifier): Promise<IUserDataProfile> {
```

Comment:
No need for the `await` here since the method returns a `Promise`?

Reply:
True. Oversight from my side